### ISSUE TITLE
ScaleMonitor is constantly logging "An item with the same key has already been added." errors

### ISSUE THREAD

**Issue by jbarntish** [microsoft/durabletask-netherite#278]
Hello,

In a Durable Function stack running Netherite that is setup on an EP2 in a 4 node scale out, we are seeing a _constant_ stream of dictionary key errors coming out of `ScaleManager`:

`An item with the same key has already been added. Key: durabletasktrigger-netherite-foo` where `foo` is the task hub name.

```
assembly: Microsoft.Azure.WebJobs.Host, Version=3.0.37.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
fileName: D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Scale\ScaleManager.cs
level: 2
line: 113
method: Microsoft.Azure.WebJobs.Host.Scale.ScaleManager+<GetScaleMonitorsResultAsync>d__11.MoveNext

customDimensions: 
{
    "HostInstanceId": "f04147c1-80de-4ed1-843e-12636ce1b896",
    "Category": "Microsoft.Azure.WebJobs.Host.Scale.ScaleManager",
    "prop__{OriginalFormat}": "Failed to query scale status for monitor 'durabletasktrigger-netherite-foo'.",
    "LogLevel": "Error",
    "ProcessId": "24",
    "FormattedMessage": "Failed to query scale status for monitor 'durabletasktrigger-netherite-foo'."
}
```

After searching the `Microsoft.Azure.WebJobs` github I can see this is being logged as an error in this code bock in [ScaleManager.cs](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Scale/ScaleManager.cs):

```
              _logger.LogDebug($"Monitor '{monitor.Descriptor.Id}' voted '{result.Vote.ToString()}'");
                        string key = monitor.Descriptor.FunctionId ?? monitor.Descriptor.Id;
                        votes.Add(key, new ScaleStatus()
                        {
                            Vote = result.Vote
                        });
                    }
                    catch (Exception exc) when (!exc.IsFatal())
                    {
                        // if a particular monitor fails, log and continue
                        _logger.LogError(exc, $"Failed to query scale status for monitor '{monitor.Descriptor.Id}'.");
                    }
```

Stepping one level up to the DurableTask framework, I checked for any existences of the key `durabletasktrigger` and found some, however, all of their instances are appending function ids to it: 

`return new NoOpScaleMonitor($"{functionId}-DurableTaskTrigger-{this.Options.HubName}".ToLower());`

Going one more level up to Netherite, I found another instance of the `durabletasktrigger` key. The function id is not being appended to this one:

```
public ScaleMonitor(ScalingMonitor scalingMonitor)
            {
                this.scalingMonitor = scalingMonitor;
                this.descriptor = new ScaleMonitorDescriptor($"DurableTaskTrigger-Netherite-{this.scalingMonitor.TaskHubName}".ToLower());
            }
```
At this point, I do not see any major issues with our process. It seems the error is logged and things move on. That said, it's a _constant_ stream of these; we've seen this error in exception logs 8k times in the last hour alone. 

Any thoughts on this? Is there any reason to be concerned scaling is being affected? I realize the WebJob's framework is just logging and moving on, so maybe it's not a huge deal.

Please let me know if I can provide any additional detail.

**Comment by sebastianburckhardt** [comment]
I think this error is a consequence of a change in the scale monitor runtime which now complains if multiple scale monitors have the same id. I do not think it impacts function since at least one scale monitor is still getting registered and should be able to do what it needs to do.  

I believe @davidmrdavid has already fixed this in PR #270, so these errors should disappear with the next release.

**Comment by sebastianburckhardt** [comment]
The latest release [1.4.0](https://github.com/microsoft/durabletask-netherite/releases/tag/1.4.0) should fix this.

If not, let me know and I will take another look.

**Comment by jbarntish** [comment]
Just upgraded to 1.4.0, looks like it's still occurring.

**Comment by rafallopatka** [comment]
Confirming, this error still occurs.

**Comment by sebastianburckhardt** [comment]
This should be fixed with the latest release (1.4.1)

**Comment by sebastianburckhardt** [comment]
This was fixed in the latest release (1.4.1)
